presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java

```diff
@@ -662,4 +702,335 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
+        }
+
+        Map<String, Object> finalProperties = new HashMap<>();
+        finalProperties.put(FILE_FORMAT_PROPERTY, FileFormat.PARQUET);
+        if (definition.getProperties().containsKey(PARTITIONING_PROPERTY)) {
+            finalProperties.put(PARTITIONING_PROPERTY, definition.getProperties().get(PARTITIONING_PROPERTY));
+        }
+
+        SchemaTableName storageTable = new SchemaTableName(viewName.getSchemaName(), storageTableName);
+        ConnectorTableMetadata tableMetadata = new ConnectorTableMetadata(storageTable, columns, finalProperties, Optional.empty());
+
+        Optional<ConnectorNewTableLayout> layout = getNewTableLayout(session, tableMetadata);
+        finishCreateTable(session, beginCreateTable(session, tableMetadata, layout), ImmutableList.of(), ImmutableList.of());
+
+        tableBuilder.setViewOriginalText(Optional.of(encodeMaterializedViewData(definition)))
+                .setViewExpandedText(Optional.of("/* Presto Materialized View */"));
+        Table table = tableBuilder.build();
+        PrincipalPrivileges principalPrivileges = buildInitialPrivilegeSet(session.getUser());
+        if (existing.isPresent()) {
+            if (!replace) {
+                throw new MaterializedViewAlreadyExistsException(viewName);
+            }
+            // drop the current storage table
+            if (existing.get().getParameters().containsKey(STORAGE_TABLE)) {
+                String currentStorageTable = existing.get().getParameters().get(STORAGE_TABLE);
+                metastore.dropTable(identity, viewName.getSchemaName(), currentStorageTable, true);
+            }
+            metastore.replaceTable(identity, viewName.getSchemaName(), viewName.getTableName(), table, principalPrivileges);
+            return;
+        }
+
+        metastore.createTable(identity, table, principalPrivileges);
+    }
+
+    @Override
+    public void dropMaterializedView(ConnectorSession session, SchemaTableName viewName)
+    {
+        final HiveIdentity identity = new HiveIdentity(session);
+        Optional<Table> view = metastore.getTable(identity, viewName.getSchemaName(), viewName.getTableName());
+        if (!view.isPresent()) {
+            throw new MaterializedViewNotFoundException(viewName);
+        }
+
+        if (view.get().getParameters().containsKey(STORAGE_TABLE)) {
+            String storageTableName = view.get().getParameters().get(STORAGE_TABLE);
+            try {
+                metastore.dropTable(identity, viewName.getSchemaName(), storageTableName, true);
+            }
+            catch (Exception e) {
+                // ignore the exception
+            }
+        }
+        metastore.dropTable(identity, viewName.getSchemaName(), viewName.getTableName(), true);
+    }
+
+    private Configuration getConfiguration(ConnectorSession session, String schemaName)
+    {
+        return hdfsEnvironment.getConfiguration(new HdfsContext(session, schemaName), new Path("file:///tmp"));
```

Comment:
This needs to use the correct `Path` for the file we are going to open. The purpose of `HdfsEnvironment.getConfiguration()` is to create a `Configuration` object with the right values for the file system we are going to use. If we don't

Reply:
Your comment cut off. But next comments talks about using HdfsInputFile. That change makes the `getConfiguration` unused, so removed the function.